docs/use-cases.rst

```diff
@@ -0,0 +1,668 @@
+Use Cases
+=========
+
+.. include:: _include.rst
+
+The following use cases represent the kinds of interaction users may have with
+Truffle and other tools. These use cases serve to help validate proposals for
+changes to Truffle's artifact format.
+
+These use cases are written primarily for Truffle, as the primary producer
+and consumer of artifacts, but should apply generally for any and all
+tooling for smart contract / decentralized application development.
+
+Since the process to develop and deploy a decentralized applications can
+include many participants, these use cases refer to several different user
+profiles:
+
+- Smart Contract Developer
+- Frontend Developer
+- Business Stakeholder
+- Auditor
+
+Not all of these users interact with development tooling directly, or in the
+same ways. As such, these use cases should specify appropriately.
+
+This list may not be exhaustive, but should cover the vast majority of required
+behaviors.
+
+
+Saving artifacts
+----------------
+
+Users and tooling must be able to save new and update existing |Artifact|
+files.
+
+As a core behavior, tooling must facilitate the internal record-keeping of all
+known |ContractTypes| and |ContractInstances|, updating them appropriately
+when they change, or when knowledge of them changes.
+
```

Comment:
Do you have something specific in mind with this distinction between the actual and the known?

Reply:
Errr... not a good enough distinction to be able to explain it right now. Basically I'm thinking that the tool can effect some change, and that modifies the contract artifacts, or something else can change (outside the tool, e.g. a new version of a public library), and the artifacts have to get updated accordingly.

Reply:
Ah yes! That's a good distinction I think!

Reply:
Should I add it back in? I wasn't happy with how unclear it was

Reply:
I don't know - now that you've explained it me I feel completely satisfied.